Implement variadic function (#315)
JuliaInterop/Clang.jl: docs/src/generator.md

# Generator Tutorial

## Variadic Function
With help of `@ccall` macro, variadic C functions can be from Julia, for example, `@ccall printf("%d\n"::Cstring; 123::Cint)::Cint` will call the C function `printf`. Note the semicolon `;` indicates variadic argument.

> If `wrap_variadic_function` in `codegen` section of options is set to true, variadic functions are also wrapped as:

Reviewer: ```suggestion
If `wrap_variadic_function` in `codegen` section of options is set to `true`, `Clang.jl` will generate wrappers for variadic C functions, for example, `printf` will be wrapped as:
```

```julia
@generated function printf(fmt, va_list...)
        :(@ccall(libexample.printf(fmt::Ptr{Cchar}; $(to_c_type_pairs(va_list)...))::Cint))
    end
```

> and can be called like normal functions without specifying types. the example above would be `LibExample.printf("%d\n", 123)`.

Reviewer: ```suggestion
It can be called just like normal Julia functions without specifying types: `LibExample.printf("%d\n", 123)`.
```
